src/random_test.cpp

```diff
@@ -41,17 +47,55 @@ static std::chrono::system_clock::time_point start_time =
 std::atomic<size_t> table_started(0);
 std::atomic<size_t> table_completed(0);
 std::atomic_flag lock_stream = ATOMIC_FLAG_INIT;
-std::atomic<bool> connection_lost(false);
+std::atomic<bool> run_query_failed(false);
+/* partition type supported by system */
 std::vector<Partition::PART_TYPE> Partition::supported;
 
 /* get result of sql */
-static MYSQL_RES *get_result(std::string sql, Thd1 *thd) {
+static bool get_check_result(const std::string &sql, Thd1 *thd) {
+
+  bool success = true;
+  thd->result = nullptr;
   thd->store_result = true;
+
   execute_sql(sql, thd);
-  auto result = thd->result;
+
+  if (thd->result == nullptr)
+    success = false;
+
+  /* Print error message thread logs
+  Displays just the contents of first row in case of an
+  error. Remove, for more details/rows. */
+  if (success) {
+    auto row = mysql_fetch_row(thd->result);
+    // Check the value of Msg_text column
+    if (strcmp(row[3], "OK") != 0) {
+      thd->thread_log << "Error: " << row[0] << " " << row[1] << " " << row[2]
+                      << " " << row[3] << std::endl;
+      success = false;
+    }
+  }
+
+  mysql_free_result(thd->result);
+  thd->result = nullptr;
+  thd->store_result = false;
+  return success;
+}
```

Comment:
Good change to avoid code repetition to check for "OK". But, get_result was a generalized method which executed a query and returned the result set of the query. So, instead of modifying it, I think get_check_result should use get_result internally and do it's specific check since the general get_result method could be useful in the future.

Reply:
Hello Varun, The have `mysql_read_single_value`  generalized method to get the result set of the query.

Reply:
mysql_read_single_value returns only a single value and not the entire result set which could be useful in some cases. Anyway, I guess it could be implemented in the future when needed.